### DISCUSSION TITLE
V5: provide an easy way to get node config from snapshot

### DISCUSSION THREAD

**Discussion by christophe-g** [statelyai/xstate#4511] [Ideas] [answered]
There might already be better ways to do this, but could not figure out from the docs and the current codebase. 

#### Usecase
In a cloud environment, you wish to be able to stop your actors (and cloud invocation) as soon as possible. This would be for instance, as soon as a machine reaches: 

1. status is `final` or `error` , 
2. or a state that has transitions (and is not `always`) - sending an event would revive and restore the actor so that it continues its live cycle.
3. _... potentially for later: has all its children in 1. or 2._

**1.** is easy enough. 

For **3.** ended up with: 
```ts
await waitFor(this.actor!, (snap) => {
  // const nodeConfig = this.machine.getStateNodeById(`${this.machine.id}.${snap.toStrings().pop()}`)
  const nodeConfig = this.machine.getStateNodeById(`${this.machine.id}.${snap.value}`)
  return snap.status === 'done'
    || snap.status === 'error'
    || (nodeConfig.transitions.size !== 0 && !nodeConfig.always)
})

```

- Would it make sense to add a more robust  `getNodeConfig` for the snapshot API ? 
- Would it make sense to add a new snapshot attribute or method that indicates whether the state will only transition on receiving a new event (smth like `snapshot.isTransient()`)?

Btw - but I am sure you already know that - the `snapshot.toStrings`  method does not seem to be implemented.

Thanks anyway.

> P.S. this is my last question for this week...

**Comment by davidkpiano** [comment, marked as answer]
You can access the nodes in a simpler way:

```ts
const nodes = snapshot._nodes;
```

**Reply by christophe-g** [reply]
Thank you !
